spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkReadConf.java

```diff
@@ -83,7 +91,11 @@ public Long endSnapshotId() {
   }
 
   public String branch() {
-    return confParser.stringConf().option(SparkReadOptions.BRANCH).parseOptional();
+    String optionBranch = confParser.stringConf().option(SparkReadOptions.BRANCH).parseOptional();
+    ValidationException.check(
```

Comment:
This check ensures that we discourage the use of read option to avoid further confusion. Technically we can allow this when the option branch and identifier branch are the same, but that is discouraged anyway so I decided to block them all together.